### ISSUE TITLE
Improve FreeCAD Hidden Line Removal

### ISSUE THREAD

**Issue by WandererFan** [opencax/GSoC#69]
## Outline
FreeCAD's Technical Drawing module (TechDraw) relies heavily on the OpenCascade Hidden Line Removal algorithms.  These algorithms can be very slow, do not provide progress reporting and do not provide any linkage between the input shape and the output.

## Details
The TechDraw module provides projections, section views and detail views of 3D model components and assemblies developed in FreeCAD modules such as Part, PartDesign and Draft.

## Expected Outcome
a) develop new code for projecting shapes and creating the geometry for technical drawings. 
-or-
b) modify the existing OpenCascade code as an enhancement.

## Project Properties
Both OpenCascade and TechDraw are written in C++.

### Skills
The student should have a good knowledge of C++ and be familar with graphics topics such as the painters algorithm, face detection and hidden line removal.
Knowledge of technical drawing standards and previous exposure to Qt will be helpful.  Familiarity with OpenCascade is a definite plus.

### Difficulty 
Hard

### Size 
long

### Additional Information

*  Potential mentor(s): wandererfan
*  Organization website: https://freecadweb.org
*  Communication channels: https://forum.freecadweb.org

**Comment by aothms** [comment]
Sounds good to me! We don't have a lot of mentors on our side with deep knowledge of C++ open cascade so we have to balance this with what else is coming in, but other than that, sure!

**Comment by yorikvanhavre** [comment]
I think this is no problem, we can do most of the mentoring on the FreeCAD side I guess.

**Comment by yorikvanhavre** [comment]
@brlcad I could transfer all the FreeCAD issues to the FreeCAD repo, so it does not creates unnecessary confusion for students... Ok for you?

**Comment by yorikvanhavre** [comment]
@brlcad I'll transfer all the remaining FreeCAD issues to FreeCAD's repo, except for those that are not 100% FreeCAD, if that's OK for you? So we don't create confusion for the potential interested...

**Comment by brlcad** [comment]
@yorikvanhavre Sounds good!  I already closed many (most?) of them out simply because I saw you had them listed on your wiki project ideas page.  I kept the two joint IFCOpenShell ones open.  You're also of course welcome to keep using this space for FreeCAD's ideas too, if you want.  It would be good to have some space for continuing to exchange ideas, discussions, and share code..

**Comment by yorikvanhavre** [comment]
Definitely! I'll link back everything to here anyway, so we keep enabling potential cross-project projects...

**Comment by ghost** [comment]
Hi @yorikvanhavre @WandererFan , wanted to enquire if is this project there in GSoC'25 list.
Thanks!! Would like to understand more on this project if any documentation and resources are there to see!!

**Comment by yorikvanhavre** [comment]
This issue is now migrated to https://github.com/FreeCAD/FreeCAD/issues/19442

Yes, it will be part of GSoC2025. It is a complex issue that needs fairly good understanding of OCCT. Unfortunately there is no ready documentation for it...
